sogs/model/message.py

```diff
@@ -42,6 +42,25 @@ def __init__(self, row=None, *, sender=None, recip=None, data=None):
         assert row is not None
         self._row = row
 
+    @staticmethod
+    def delete_all(*, recip=None, sender=None):
+        """Delete all messages sent to a user or from a user.
+        returns the number of rows affected.
+        """
+        assert recip or sender
```

Comment:
This should throw an exception rather than a debug assert.  But also, given the query construction below, it needs to be the case that the caller has specified exactly one of these, rather than at least one: otherwise this function is going to generate an invalid query.